### PR TITLE
ColorPicker Updates

### LINKED ISSUE #11059: ColorPicker Hue Slider not displayed correctly
**Describe the bug**
The background of the hue slider of ColorPicker is not displayed correctly.
It should not change according to selected color, and should keep bright and saturated.

**To Reproduce**
1. Create a ColorPicker with SaturationValue component
2. Change the color
3. See the background of the Hue slider
4. It is changing, but this is not the normal behavior of such control.
5. It can even become pure black or pure white which is not correct.
6. The correct rendering is the one when the selected color is at top-right location.

**Expected behavior**
The Hue slider should always look like that:
image

**Screenshots**
image

**Desktop (please complete the following information):**

- OS: Windows 10
- Version 11 preview 6

### PR DESCRIPTION
**Pull request by robloo**
## What does the pull request do?

Updates the ColorPicker group of controls with more fixes/improvements.

 1. A special case was added to handle #00000000 in the ColorSpectrum upon first selection. If this case is detected the third component and alpha component will automatically be set to maximum values. Full reasoning is commented in code.
 1. ColorSlider.`IsAlphaMaxForced` was renamed to `IsAlphaVisible`. This follows terminology in other controls/primitives and essentially does the same as before.
 1. One of the largest changes is ColorSlider.`IsSaturationValueMaxForced` being replaced with the more powerful/general-purpose `IsPerceptive` property with enhanced functionality. Before this property only worked in Hsva color model, now it also works in Rgba. This allows switching the spectrum slider gradient to render as mathematically correct vs. better for user perception.
      1. By default this changes the look of the RGB color model sliders:
          image
          They now no longer change -- the gradients will always be red/green/blue in hue.
      1. Alpha slider stays the same and HSV sliders stay the same.
      1.  When this property is set should be changed in the styles/control themes. However, it doesn't appear to work #11271. The API was updated now so this can be addressed later though as a non-breaking change.
 1. Added direct conversions for HsvColor.ToHsl() and HslColor.ToHsv()
 1. Other minor improvements to comments

It was planned to adjust the ColorPreviewer accent color calculation to match the Fluent theme (HSL color model algorithm). However, there were two issues found that stopped this:
 * The algorithm is not reversible, reducing by one step and then increasing by one step will result in a different color than the start.
 * For some reason, hue information is lost when the HSL color reaches minimum/maximum lightness. In HSL this becomes black and white. This then switches the base color to pure black and white with no hue information. I'm not actually sure what is causing this as I added in full conversions between HSV/HSL without going through RGB so information shouldn't be lost. Its just something about HSL I need to think about further I guess.

cc @maxkatz6 

## How was the solution implemented (if it's not obvious)?

As described.

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- ~[ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation~

## Breaking changes

Yes, breaking ColorSlider properties from previous previews. These are so far unreleased in stable versions of Avalonia and are only used in the primitive ColorSlider control.
 1. ColorSlider.`IsAlphaMaxForced` renamed to `IsAlphaVisible`
 4. ColorSlider.`IsSaturationValueMaxForced` replaced with more powerful/general-purpose `IsPerceptive` property with enhanced functionality

## Obsoletions / Deprecations

None

## Fixed issues

Fixes #11059

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034358-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Controls.ColorPicker/Themes/Fluent/ColorPicker.xaml [hunk 1]
```diff
@@ -490,18 +490,33 @@
...
+
+    <Style Selector="^ /template/ primitives|ColorSlider#Component1Slider[ColorModel=Rgba]">
+      <Setter Property="IsPerceptive" Value="False" />
+    </Style>
+    <Style Selector="^ /template/ primitives|ColorSlider#Component2Slider[ColorModel=Rgba]">
+      <Setter Property="IsPerceptive" Value="False" />
+    </Style>
+    <Style Selector="^ /template/ primitives|ColorSlider#Component3Slider[ColorModel=Rgba]">
+      <Setter Property="IsPerceptive" Value="False" />
+    </Style>
```
Comment (robloo) [inline src/Avalonia.Controls.ColorPicker/Themes/Fluent/ColorPicker.xaml:518]: There seems to be an issue here: https://github.com/AvaloniaUI/Avalonia/issues/11271

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034364-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Controls.ColorPicker/ColorSlider/ColorSlider.Properties.cs [hunk 2]
```diff
@@ -109,14 +109,39 @@ public HsvColor HsvColor
...
-        public bool IsAlphaMaxForced
+        /// <remarks>
+        /// For example, with Hue in the HSVA color model, the Saturation and Value components are always forced
+        /// to maximum values during rendering. In the RGBA color model, all components other than
+        /// <see cref="ColorComponent"/> are forced to minimum values during rendering.
+        /// <br/><br/>
+        /// Also note this property will only adjust components other than <see cref="ColorComponent"/> during rendering.
+        /// This doesn't change the values of any components in the color – it is only for display.
+        /// </remarks>
+        public bool IsPerceptive
```
Comment (robloo) [inline src/Avalonia.Controls.ColorPicker/ColorSlider/ColorSlider.Properties.cs:141]: The term "perceptive" is used here as it indicates the control should improve "user-perception". "perceptive" perhaps doesn't make the most sense in the context of the control itself though. The ColorSlider isn't "perceptive".

If anyone has concerns with this property naming please let me know and if you have a better idea. I can't really find examples of this concept anywhere.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034606-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Controls.ColorPicker/Themes/Fluent/ColorPicker.xaml (hunk 1 above)
Comment (robloo) [reply src/Avalonia.Controls.ColorPicker/Themes/Fluent/ColorPicker.xaml:519]: Removed for now.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034672-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by robloo** [comment]
@maxkatz6 This one should be ready for review.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0034681-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by grokys — APPROVED** [review]

**Pull request merged**